features/FEATURE_LWIP/lwip-interface/lwip_stack.c

```diff
@@ -1023,6 +1043,36 @@ static nsapi_error_t mbed_lwip_setsockopt(nsapi_stack_t *stack, nsapi_socket_t h
             }
             return 0;
 
+        case NSAPI_ADD_MEMBERSHIP:
+        case NSAPI_DROP_MEMBERSHIP: {
+#if LWIP_IPV6
+            return NSAPI_ERROR_UNSUPPORTED; 
+#else
+            if (optlen != sizeof(nsapi_ip_mreq_t)) {
+                return NSAPI_ERROR_UNSUPPORTED;
+            }
+
+            err_t igmp_err;
+            nsapi_ip_mreq_t *imr = (nsapi_ip_mreq_t *)optval;
```

Comment:
Cast not required cos it's C. Its removal will encourage you to preserve const-correctness.